### PR TITLE
feat: Add helm chart

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR:

- adds the initial helm chart

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by MarcelMue — APPROVED** [review]

File: helm/cluster-api-provider-vsphere/templates/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: vsphere
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-leader-election-rolebinding
+  namespace: {{ include "resource.default.namespace" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: leader-election-role
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-vsphere/templates/rbac.yaml:192]:
```suggestion
  name: {{ include "resource.default.name" . }}-leader-election-role
```

File: helm/cluster-api-provider-vsphere/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-vsphere/templates/rbac.yaml:195]:
```suggestion
  name: {{ include "resource.default.name" . }}
```

File: helm/cluster-api-provider-vsphere/templates/rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: vsphere
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-manager-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: manager-role
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-vsphere/templates/rbac.yaml:208]:
```suggestion
  name: {{ include "resource.default.name" . }}-manager-role
```

File: helm/cluster-api-provider-vsphere/templates/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-vsphere/templates/rbac.yaml:211]:
```suggestion
  name: {{ include "resource.default.name" . }}
```

File: helm/cluster-api-provider-vsphere/templates/rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: vsphere
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-proxy-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: proxy-role
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-vsphere/templates/rbac.yaml:224]:
```suggestion
  name: {{ include "resource.default.name" . }}-proxy-role
```

File: helm/cluster-api-provider-vsphere/templates/rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (TheoBrigitte) [inline helm/cluster-api-provider-vsphere/templates/rbac.yaml:227]:
```suggestion
  name: {{ include "resource.default.name" . }}
```

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]

File: helm/cluster-api-provider-vsphere/templates/rbac.yaml [hunk 7]
```diff
@@ -0,0 +1,222 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: {{ include "resource.default.name" . }}-leader-election-role
```
Comment (kubasobon) [inline helm/cluster-api-provider-vsphere/templates/rbac.yaml:6]: FYI We got rid of all leader-election roles/rolebindings in other controllers

Comment (QuentinBisson) [reply helm/cluster-api-provider-vsphere/templates/rbac.yaml:6]: Thanks

Comment (QuentinBisson) [reply helm/cluster-api-provider-vsphere/templates/rbac.yaml]: I removed them

**Pull request merged**